libwacom/util-strings.h

```diff
@@ -132,6 +132,21 @@ safe_atoi_base(const char *str, int *val, int base)
 	return true;
 }
 
+static inline bool
+safe_itoa_base(int val, char *buffer, int buf_size, int base)
+{
+	assert(base == 10 || base == 16 || base == 8);
+
+	if (base == 10)
+		snprintf(buffer, buf_size, "0x%08d", val);
+	else if (base == 16)
+		snprintf(buffer, buf_size, "0x%08x", val);
+	else if (base == 8)
+		snprintf(buffer, buf_size, "0x%08o", val);
+
+	return true;
+}
```

Comment:
i know this feels pointless but can we add some test cases for this so we never have to worry about this again?

Arguably a `switch` on the base could look nicer but I'm not going to die on that particular hill :)